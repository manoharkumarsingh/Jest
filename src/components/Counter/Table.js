import React from "react";

function Table() {
  return (
    <table className="custom-border">
      <thead>
        <tr>
          <th>Name</th>
          <th>Email</th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Manohar</td>
          <td>mksnitc@gmail.com</td>
        </tr>
      </tbody>
    </table>
  );
}

export default Table;
