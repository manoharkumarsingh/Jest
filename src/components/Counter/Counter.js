import React from "react";
import Table from "./Table";

class Counter extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      count: 0,
    };
  }
  render() {
    const { count } = this.state;
    return (
      <div>
        <div className={`clicks-${count}`}>{count} clicks</div>
        <button
          onClick={() => {
            this.setState({ count: count + 1 });
          }}
        >
          Increment
        </button>
        <h1>Article Details</h1>
        <Table></Table>
      </div>
    );
  }
}

export default Counter;
