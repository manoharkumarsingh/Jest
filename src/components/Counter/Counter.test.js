import Counter from "./Counter";
import { shallow, mount } from "enzyme";
import Table from "./Table";

describe.skip("Counter Page Test", () => {
  it("Click count", () => {
    let wrapper = shallow(<Counter />);
    wrapper.find("button").simulate("click");
    expect(wrapper.find(".clicks-1").length).toEqual(1);
  });

  it("Click 3 count", () => {
    let wrapper = shallow(<Counter />);
    wrapper.find("button").simulate("click");
    wrapper.find("button").simulate("click");
    expect(wrapper.find(".clicks-2").length).toEqual(1);
  });
  /**Child component  */
  it("Dive", () => {
    let wrapper = shallow(<Counter />);
    expect(wrapper.find(Table).dive().find(".custom-border").length).toEqual(1);
  });

  /**Child component  */
  it("Snapshot 1", () => {
    let wrapper = shallow(<Counter />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe("Mount vs shallow", () => {
  it("Full rendering", () => {
    let wrapper = mount(<Counter />);
    // console.log(wrapper.debug());
  });
});
