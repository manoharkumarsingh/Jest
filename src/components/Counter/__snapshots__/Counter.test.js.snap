// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Page Test Snapshot 1 1`] = `
<div>
  <div
    className="clicks-0"
  >
    0
     clicks
  </div>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <h1>
    Article Details
  </h1>
  <Table />
</div>
`;

exports[`Mount vs shallow Snapshot 1 1`] = `
<div>
  <div
    className="clicks-0"
  >
    0
     clicks
  </div>
  <button
    onClick={[Function]}
  >
    Increment
  </button>
  <h1>
    Article Details
  </h1>
  <Table />
</div>
`;
